### ISSUE TITLE
wss://tracker.btorrent.xyz/ is no longer available and is triggering AVG warnings

### ISSUE THREAD

**Issue by draeder** [chr15m/bugout#16]
The following error appears consistently in the logs when using the current minified API. The reason is because the developer no longer has the webtorrent tracker server online:

`bugout.min.js:11 WebSocket connection to 'wss://tracker.btorrent.xyz/' failed: Error during WebSocket handshake: Unexpected response code: 522`

Also, for users who have AVG Free installed (potentially other AV software), that URL is triggering a _URL:Blacklist_ warning (see below screenshot). 

Might I suggest updating the Readme and/or API documentation with some suggested webtorrent trackers that could be used with the announce field per #14 , and remove hard coded trackers from the API?

![avg](https://user-images.githubusercontent.com/15527661/70112341-a61b7700-161a-11ea-99e5-9cc422bc6349.PNG)

**Comment by chr15m** [comment]
Good idea, thanks.

**Comment by draeder** [comment]
I just realized this may not be a bug in (only) your API code. I specified trackers to use and still got the warning about the dead one. So, I started playing around with webtorrent's API and I'm also getting the same console log errors in the browser, and my buddy is getting the AVG popup. It looks like this might be a bug in webtorrent, too.

**Comment by chr15m** [comment]
@draeder Bugout is supposed to pass through the trackers specified to WebTorrent:

https://github.com/chr15m/bugout/blob/master/index.js#L37-L38

**Comment by chr15m** [comment]
Here is where it passes the announce field through to WebTorrent's seed function: https://github.com/chr15m/bugout/blob/master/index.js#L77

**Comment by draeder** [comment]
@chr15m You're right -- that last observation was due to an oversight in my code. I forgot to pass the announce field in a 2nd bugout instance I was running, my apologies!

**Comment by chr15m** [comment]
@draeder all good, even so I think your suggestion regarding the documentation is a good one.

**Comment by vrboy** [comment]
Im still getting the error from AVG anti Virus free from https://webtorrent.io/

Weve safely aborted connection on tracker.btorrent.xyz because it was infected with URL:blacklist

Did anyone get a work around? Will scare off my users. Is the error still there when you use the code on your own site?

Many thanks in advance

**Comment by draeder** [comment]
@vrboy Use the answer in #14 to pass your own trackers.

**Comment by vrboy** [comment]
Many many thanks for finding that link for me Draeder. I'm not an advanced coder and new to P2P code.  Two quick questions please. 
1. Do you recommend using another tracker. I am not sure which one to use?
2. It seems I need to alter the index.js file from looking above. Is that a file I download edit then upload to my server and put the index.js into the route next to index.html? 

If you can suggest how to alter it would be much appreciated.

This is the part I have to edit only I presume?
  this.announce = opts.announce || ["wss://hub.bugout.link", "wss://tracker.openwebtorrent.com", "wss://tracker.btorrent.xyz"];
  this.wt = opts.wt || new WebTorrent({tracker: trackeropts});

I am trying to become a commercial customer if I can get this working.  Happy to pay for a fix if needed. 

Many many thanks in advance

**Comment by chr15m** [comment]
@vrboy When you instantiate your Bugout object you pass in the announce argument:

`var b = Bugout(identifier_or_room_name, {"announce": ["https://YOUR-TRACKER-HERE..."]});`

There are only a few trackers out there and if none of them work you have to run your own. This is a bad situation and it is why I am working on this: https://github.com/chr15m/webrtc-signaling-mesh

Stay tuned.

**Comment by vrboy** [comment]
Thx for the message chr15m,
I need a tracker in the next few days to launch. Any idea when will be ready please?

Two more questions guys
1. I saw a list of torrents on this page
https://www.axlcash.com/category/random-babbling
Including

**Comment by chr15m** [comment]
@vrboy you shouldn't have to edit `index.js` (if you mean the file that ships with Bugout) - see the code snippet I posted above. The webtorrent tracker is already included in the set that Bugout uses by default.

**Comment by vrboy** [comment]
If anyone has an upwork account I would gladly pay for them to work on this problem for me...

Thx for the reply chr15m. 

The index.js Listed on this page 
https://github.com/webtorrent/webtorrent#usage

is only place I can see in the code that has this line that refers to the "Virus" btorrent.xyz (as deemed by AVG and others)

this.announce = opts.announce || ["wss://hub.bugout.link", "wss://tracker.openwebtorrent.com", "wss://tracker.btorrent.xyz"];

In idiot programming terms can you tell me how do I get rid of any mention or connection to btorrent.xyz in the https://webtorrent.io/ or webtorrent.min.js so that the code wont start a virus warning in Chrome.

In other words how do I use these two trackers that work without virus warnings 
wss://tracker.novage.com.ua
wss://tracker.openwebtorrent.com

Tested here
http://novage.com.ua/p2p-media-loader/demo.html

Many Many thanks in advance

**Comment by draeder** [comment]
@vrboy I think you might be confusing the webtorrent api with the bugout api.

**Comment by chr15m** [comment]
@draeder I've pushed an update to the docs to try to clarify the issue of specifying trackers. Let me know if you get a chance to look and comment: https://github.com/chr15m/bugout/blob/master/docs/API.md#using-your-own-signaling-servers
